### PR TITLE
feat: SSE

### PR DESCRIPTION
**Pull request by JamieSinn**
SSE Connection handling

### CODE REVIEW

File: src/examples/java/com/devcycle/examples/LocalExample.java [hunk 1]
```diff
@@ -22,8 +23,13 @@ public static void main(String[] args) throws InterruptedException {
...
 
-        DevCycleLocalOptions options = DevCycleLocalOptions.builder().configPollingIntervalMs(60000)
-                .disableAutomaticEventLogging(false).disableCustomEventLogging(false).build();
+        DevCycleLocalOptions options = DevCycleLocalOptions.builder()
+                .configPollingIntervalMS(60000)
+                .disableAutomaticEventLogging(false)
+                .customLogger(new SimpleDevCycleLogger(SimpleDevCycleLogger.Level.DEBUG))
+                .disableCustomEventLogging(false)
+                .enableBetaRealtimeUpdates(true)
+                .build();
```
Comment (jonathannorris) [inline src/examples/java/com/devcycle/examples/LocalExample.java:32]: are half of these needed? won't it keep events by default?

File: src/examples/java/com/devcycle/examples/LocalExample.java [hunk 2]
```diff
@@ -46,5 +52,6 @@ public static void main(String[] args) throws InterruptedException {
         } else {
             System.out.println("feature is NOT enabled");
         }
+        Thread.sleep(10000);
```
Comment (jonathannorris) [inline src/examples/java/com/devcycle/examples/LocalExample.java:55]: why this long sleep?

File: src/main/java/com/devcycle/sdk/server/common/interceptor/CustomHeaderInterceptor.java [hunk 3]
```diff
@@ -23,7 +23,7 @@ public CustomHeaderInterceptor(IRestOptions restOptions) {
     public Response intercept(Chain chain) throws IOException {
         Request request = chain.request();
 
-        if (restOptions != null) {
+        if (restOptions != null && chain.request().url().url().getHost().contains("devcycle")) {
```
Comment (jonathannorris) [inline src/main/java/com/devcycle/sdk/server/common/interceptor/CustomHeaderInterceptor.java:26]: what if we move the SSE connection to our domain?

Comment (JamieSinn) [reply src/main/java/com/devcycle/sdk/server/common/interceptor/CustomHeaderInterceptor.java:26]: This was from testing, need to remove

File: src/examples/java/com/devcycle/examples/LocalExample.java (hunk 1 above)
Comment (JamieSinn) [reply src/examples/java/com/devcycle/examples/LocalExample.java:32]: This is just reformatted from what it was before

File: src/examples/java/com/devcycle/examples/LocalExample.java (hunk 2 above)
Comment (JamieSinn) [reply src/examples/java/com/devcycle/examples/LocalExample.java:55]: To actually connect to SSE and pull a config

File: src/examples/java/com/devcycle/examples/LocalExample.java (hunk 1 above)
Comment (jonathannorris) [reply src/examples/java/com/devcycle/examples/LocalExample.java:32]: can we cleanup the ones here that are just using the default values?

**Review by jonathannorris — APPROVED** [review]

**Pull request merged**